buildbot/master/files/config/factories.py

```diff
@@ -110,6 +110,22 @@ def make_step(self, command):
                 step_kwargs['logEnviron'] = False
                 step_env += envs.upload_nightly
 
+        step_desc_suffix = []
```

Comment:
I'd prefer to integrate this code into the loop above instead of trying to re-parse things like `--log-.*`. Here's a rough sketch of one way to do it, feel free to modify as needed for ergonomics:
- Create a `step_desc` list before the `for arg in args:` loop.
- Inside the `./mach` block, append both the `./mach` token and the `mach_arg` token to `step_desc`. 
- Don't change the `log` block, so those tokens won't get added to the description.
- Add an `else` or similar at the end to append any other tokens to the description. (I think you might have to change the last `elif` to accommodate this properly.)